src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -281,6 +281,25 @@ internal static bool IsBinary(string text)
         public abstract int Length { get; }
 
         /// <summary>
+        /// The size of the storage representation of the text (in characters).
+        /// This can differ from length when storage buffers are reused to represent fragments/subtext.
+        /// </summary>
+        internal virtual int Size
```

Comment:
Is there ever a case where Size is not equal to StorageKey.Length? Otherwise it might be easier to simply have a single virtual property and then Size can just be implemented that way. I'm a bit worried that this is error prone: right now if you were to override one but not the other all sorts of weirdness might happen.